src/v/cloud_topics/level_zero/stm/ctp_stm_api.cc

```diff
@@ -40,7 +39,8 @@ std::ostream& operator<<(std::ostream& o, ctp_stm_api_errc errc) {
 }
 
 ctp_stm_api::ctp_stm_api(ss::shared_ptr<ctp_stm> stm)
-  : _stm(std::move(stm)) {}
+  : _stm(std::move(stm))
+  , _log(_stm->log()) {}
```

Comment:
my hunch is that there is something going on with lifetime here. before we had the always-available cd_log, and now stm->log needs to (1) be constructed before and (2) outlive the ctp stm api. this seems to just be happening in a unit test, so maybe there is something problematic in the test?